tests/PsrLogAdapterTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace MongoDB\Tests;
+
+use MongoDB\Driver\Monitoring\LogSubscriber;
+use MongoDB\Exception\UnexpectedValueException;
+use MongoDB\PsrLogAdapter;
+use PHPUnit\Framework\TestCase as BaseTestCase;
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+
+use function func_get_args;
+use function MongoDB\Driver\Monitoring\mongoc_log;
+use function sprintf;
+
+class PsrLogAdapterTest extends BaseTestCase
+{
+    private LoggerInterface $logger;
+
+    public function setUp(): void
+    {
+        $this->logger = $this->createTestPsrLogger();
+
+        PsrLogAdapter::addLogger($this->logger);
+    }
+
+    public function tearDown(): void
+    {
+        PsrLogAdapter::removeLogger($this->logger);
+    }
+
+    public function testLog(): void
+    {
+        /* This uses PHPC's internal mongoc_log() function to write messages
+         * directly to libmongoc. Those messages are then relayed to
+         * PsrLogAdapter and forwarded to each registered PSR logger.
+         *
+         * Note: it's not possible to test PsrLogAdapter::log() with an invalid
+         * level since mongoc_log() already validates its level parameter. */
```

Comment:
Although I ended up removing the previous test that called `log()` with invalid levels, it was adapted for `testWriteLogWithInvalidLevel()` below.